include/filters/filter_chain.hpp

```diff
@@ -70,30 +70,22 @@ load_chain_config(
   const std::string norm_param_prefix = impl::normalize_param_prefix(param_prefix);
 
   // Read parameters for filter1..filterN
-  for (size_t filter_num = 1; filter_num > found_filters.size(); ++filter_num) {
+  for (size_t filter_num = 1; ; ++filter_num) {
     // Parameters in chain are prefixed with 'filterN.'
     const std::string filter_n = "filter" + std::to_string(filter_num);
 
-    // Declare filterN.name and filterN.type
+    // Declare filterN.name
     rcl_interfaces::msg::ParameterDescriptor name_desc;
     name_desc.name = norm_param_prefix + filter_n + ".name";
     name_desc.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
     name_desc.read_only = false;
-    // Must be dynamically typed because later we undeclare it and redeclare
-    // it read-only, but statically typed params cannot be undeclared.
-    name_desc.dynamic_typing = true;
+    node_params->declare_parameter(name_desc.name, rclcpp::PARAMETER_STRING);
     rcl_interfaces::msg::ParameterDescriptor type_desc;
+    // Declare filterN.type
     type_desc.name = norm_param_prefix + filter_n + ".type";
     type_desc.type = rcl_interfaces::msg::ParameterType::PARAMETER_STRING;
     type_desc.read_only = false;
-    // Must be dynamically typed because later we undeclare it and redeclare
-    // it read-only, but statically typed params cannot be undeclared.
-    type_desc.dynamic_typing = true;
-
-    node_params->declare_parameter(
-      name_desc.name, rclcpp::ParameterValue(), name_desc);
-    node_params->declare_parameter(
-      type_desc.name, rclcpp::ParameterValue(), type_desc);
```

Comment:
By not passing in the entire `name_desc` or `type_desc`, don't you lose the read-only setting? (in the original code the parameters were read only; after your changes i think they are modifiable). Is that intentional?

Reply:
Nevermind... i misread  - looks like it defaults to read only, which is what the old code did.